src/MongoDB/Driver/Manager.cpp

```diff
@@ -522,6 +535,158 @@ static bool hippo_mongo_driver_manager_apply_ssl_opts(mongoc_client_t *client, c
+	o_event->o_set(s_MongoDriverMonitoringCommandResultEvent_durationMicros, Variant(mongoc_apm_command_succeeded_get_duration(event)), s_MongoDriverMonitoringCommandResultEvent_className);
+
+	/* Add the reply */
+	Variant v_reply;
+	const bson_t *b_reply = mongoc_apm_command_succeeded_get_reply(event);
+	hippo_bson_conversion_options_t options = HIPPO_TYPEMAP_INITIALIZER;
+	BsonToVariantConverter convertor(bson_get_data(b_reply), b_reply->len, options);
+	convertor.convert(&v_reply);
+	o_event->o_set(s_MongoDriverMonitoringCommandSucceededEvent_reply, v_reply, s_MongoDriverMonitoringCommandSucceededEvent_className);
+
+	TypedValue args[1] = { *(Variant(o_event)).asCell() };
+	
+	g_context->invokeFuncFew(
+		result.asTypedValue(),
+		m, obj_context,
+		nullptr, 1, args
+	);
+}
+
+
+static void command_failed(const mongoc_apm_command_failed_t *event)
+{
+	Class *cls = Unit::lookupClass(s_MongoDBDriverManager__Dispatcher.get());
+	Func  *m   = cls->lookupMethod(s_MongoDBDriverManager_dispatchCommandFailed.get());
+	static Class *c_event;
+	Variant    result;
+	ObjectData *obj_context = (ObjectData*) mongoc_apm_command_failed_get_context(event);
+	MongoDBDriverManagerData* data = Native::data<MongoDBDriverManagerData>(obj_context);
+
+	auto prop = cls->lookupSProp(s_MongoDBDriverManager_subscribers.get());
+	auto val  = cls->getSPropData(prop);
+
+	if (!val) {
+		return;
+	}
+
+	Array subscribers = tvAsVariant(val).toArray();
+
+	if (subscribers.size() < 1) {
+		return;
+	}
+
+	c_event = Unit::lookupClass(s_MongoDriverMonitoringCommandFailedEvent_className.get());
+	Object o_event = Object{c_event};
+	o_event->o_set(s_MongoDriverMonitoringCommandEvent_commandName, Variant(mongoc_apm_command_failed_get_command_name(event)), s_MongoDriverMonitoringCommandEvent_className);
+	o_event->o_set(s_MongoDriverMonitoringCommandEvent_server, Variant(hippo_mongo_driver_server_create_from_id(data->m_client, mongoc_apm_command_failed_get_server_id(event))), s_MongoDriverMonitoringCommandEvent_className);
+	o_event->o_set(s_MongoDriverMonitoringCommandEvent_operationId, Variant(mongoc_apm_command_failed_get_operation_id(event)), s_MongoDriverMonitoringCommandEvent_className);
+	o_event->o_set(s_MongoDriverMonitoringCommandEvent_requestId, Variant(mongoc_apm_command_failed_get_request_id(event)), s_MongoDriverMonitoringCommandEvent_className);
+	o_event->o_set(s_MongoDriverMonitoringCommandResultEvent_durationMicros, Variant(mongoc_apm_command_failed_get_duration(event)), s_MongoDriverMonitoringCommandResultEvent_className);
+
+	/* Add the error / exception */
+	bson_error_t b_error;
+	mongoc_apm_command_failed_get_error(event, &b_error);
+	o_event->o_set(s_MongoDriverMonitoringCommandFailedEvent_error, MongoDriver::Utils::throwExceptionFromBsonError(&b_error), s_MongoDriverMonitoringCommandFailedEvent_className);
```

Comment:
Nothing to fix here, but it's a bit odd that the method is called `throwExceptionFromBsonError()` despite only being a factory method that constructs a new exception.